### DISCUSSION TITLE
upstage-llama-2-70b-instruct-v2.ggmlv3.q5_K_M same speed llama.cpp and llama.cpp HF?

### DISCUSSION THREAD

**Discussion by viperwasp** [oobabooga/text-generation-webui#3409] [General]
Does a model need to be trained to be faster on HF for it to be faster? My results are the same. RTX 4080. 12700K running on 8 threads. Using GPU acceleration 15 layers on GPU. VRAM use 93%. 

So GPU acceleration seems to be working (BLAS = 1) on both llama.cpp and llama.cpp HF. I think I have it configured correctly. 

llama.cpp
Output generated in 156.20 seconds (0.94 tokens/s, 147 tokens, context 67, seed 896543280)

llama.cpp HF
Output generated in 98.85 seconds (1.00 tokens/s, 99 tokens, context 66, seed 399534863)

Yes I know 1.00 tokens is "faster" but trust me that is just variance. 

P.S 
Assuming HF was faster? Are there known downsides with HF like precision/accuracy does it make the results any worse?

**Comment by oobabooga** [comment]
HF will make the sampling identical to that of transformers models. I don't know if that's better or worse, but at least it gives you access to more sampling parameters and more tokenizer options, like "auto_max_new_tokens" and "add_bos_token".

**Reply by viperwasp** [reply]
Thanks I only barely understand what that means. I barely know what transformers are. But I get what your saying overall. Thank you.
